app/javascript/src/Dashboard/components/PotentialUpgradesWidget.jsx

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+import React from 'react'
+
+import { createReactWrapper } from 'utilities'
+import pluralize from 'pluralize'
+
+import 'Dashboard/styles/dashboard.scss'
+
+type Props = {
+  violations: Array<{
+    id: number,
+    account_id: number,
+    account_name: string,
+    alerts_count: number,
+    url: string
+  }>,
+  incorrectSetUp: boolean,
+  links: {
+    adminServiceApplicationPlans: string,
+    settingsAdminService: string
+  }
+}
+
+const PotentialUpgradesWidget = ({ violations, incorrectSetUp, links }: Props): React.Node => {
+  return (
+    <article className='DashboardWidget' id='dashboard-widget-potential_upgrades'>
+      <h1 className='DashboardWidget-title'>
+        Potential Upgrades
+      </h1>
+      <span className='DashboardWidget-intro'>
+        Accounts that hit their Usage Limits in the last 30 days
+      </span>
+
+      {
+        violations.some((violation) => violation)
+          ? <ol className='DashboardWidgetList'>
+            {
+              violations.map(violation =>
+                <li className='DashboardWidgetList-item' key={violation.account_id}>
+                  <a href={violation.url} className='DashboardWidgetList-link'>
+                    {violation.account_name} has {violation.alerts_count} { pluralize('Usage Limit alert', violation.alerts_count) }
+                  </a>
+                </li>
+            )}
+          </ol>
+          : (incorrectSetUp)
+            ? <>
```

Comment:
No need for a fragment here, wrap with parenthesis instead.

Reply:
I tried the parenthesis out but I got some nasty errors...

Reply:
I see. In any case, this JSX is a bit hard to read and I would extract a couple of components from there. For instance I would extract the item into a `PotentialUpgradeWidgetItem` component.

Reply:
Changed in b89bf8e.